src/t_env/env_add.c

```diff
@@ -94,6 +103,6 @@ void	env_add_or_update(t_elst *lst, char *key, char *value)
 		}
 		current = current->next;
 	}
-	new_node = env_newnode(key, value);
+	new_node = env_newnode(ft_strdup(key), ft_strdup(value));
```

Comment:
`env_newnode`
This function must be used with a function that can be malloc'd. 
I'm not sure if this is a good idea. I've thought about it a lot and this is the best I've come up with so far.